pinot-common/src/thrift/request.thrift

```diff
@@ -155,6 +155,7 @@ struct BrokerRequest {
  16: optional HavingFilterQueryMap havingFilterSubQueryMap;
  17: optional query.PinotQuery pinotQuery;
  18: optional list<SelectionSort> orderBy;
+ 19: optional i32 limit = 0;
```

Comment:
Why do we need to add this, we already support limit, right?

Reply:
The LIMIT is currently only supported in SELECT query path (BrokerRequest.setSelection()) has the LIMIT info (both start and end). Reusing this would have then led to change code down the line where existing code makes decisions based on the fact

(1) if brokerRequest.getSelections() is not null then it is guaranteed to be selection query
(2) if broerRequest.getAggregationInfo() is not null then it is guaranteed to be aggregation query

If we end up reusing the LIMIT stashed away by the parser in SelectAstNode for DISTINCT query as well then the above assumptions will break -- couple of places where existing code is dependent on these invariants are places like CombineService, BrokerReduceService

Reply:
Yes. If the user didn't specify LIMIT N, then we use the default 10 and set it here.